src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,175 @@
+            tail = newNode;
+            newNode.prev.next = newNode;
+        } else {
+            checkException(index);
+            Node finderNode = finderNode(index);
+            newNode.next = finderNode;
+            newNode.prev = finderNode.prev;
+            finderNode.prev = newNode;
+            newNode.next.prev = newNode;
+            newNode.prev.next = newNode;
+        }
+        size++;
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkException(index);
+        Node finderNode = finderNode(index);
+        return (T) finderNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> newNode = new Node<>(null, value, null);
+        T returnValue = get(index);
+
+        if (index == 0) {
+            head.next.prev = newNode;
+            newNode.next = head.next;
+            head = newNode;
+        } else {
+            Node finderNode = finderNode(index);
+            finderNode.next.prev = newNode;
+            finderNode.prev.next = newNode;
+        }
+
+        return returnValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkException(index);
+        T returnValue = get(index);
+        Node deleteNode = finderNode(index);
+        deleterNode(deleteNode, index);
+        return returnValue;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Boolean elementFound = false;
+        Node<T> deleteNode = head;
+        int index = findingIndexByValue(deleteNode, object);
+        for (int i = 0; i < size; i++) {
+            if (deleteNode.value == null && object != null) {
+                deleteNode = deleteNode.next;
+            } else if ((deleteNode.value == null && object == null)
+                    || (deleteNode.value.equals(object))) {
+                deleterNode(deleteNode, index);
+                elementFound = true;
+                break;
+            } else {
+                deleteNode = deleteNode.next;
+            }
+        }
+        return elementFound;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node finderNode(int index) {
+        Node finderNode = head;
+        for (int i = 0; i < index; i++) {
+            finderNode = finderNode.next;
+        }
+        return finderNode;
+    }
+
+    private void checkException(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Exception in thread \"main\" "
+                    + "java.lang.IndexOutOfBoundsException");
+        }
+    }
+
+    private void deleterNode(Node node, int index) {
```

Comment:
1. bad method name
2. rewrite without `index` parameter
```suggestion
    private void unlink(Node node) {
```